packages/@angular/cli/models/command.ts

```diff
@@ -0,0 +1,95 @@
+import { Logger } from '@angular-devkit/core/src/logger';
+const { cyan } = require('chalk');
+
+export interface CommandConstructor {
+  new(context: CommandContext, logger: Logger): Command;
+  aliases: string[];
+  scope: CommandScope.everywhere;
+}
+
+export enum CommandScope {
+  everywhere,
+  inProject,
+  outsideProject
+}
+
+export abstract class Command {
+  constructor(context: CommandContext, logger: Logger) {
+    this.logger = logger;
+    if (context) {
+      this.project = context.project;
+      this.ui = context.ui;
+    }
+  }
+
+  initialize(_options: any): Promise<void> {
+    return Promise.resolve();
+  }
+
+  validate(_options: any): boolean | Promise<boolean> {
+    return true;
```

Comment:
No need for this, right?

Reply:
It's a base implementation that can be overridden, but the runner relies on a boolean being returned. Not all commands will implement validate.

Reply:
Sorry I pointed to the wrong file. the Doc command still has an empty override.